packages/pigeon/lib/swift_generator.dart

```diff
@@ -43,6 +43,44 @@ class SwiftOptions {
 /// Calculates the name of the codec that will be generated for [api].
 String _getCodecName(Api api) => '${api.name}Codec';
 
+/// Returns a new [Method] using the swift function signature from [func],
+/// ie the name of function e the strings between the semicolons.
+/// Example:
+///   _swiftMethod('@SwiftFunction('add(_:with:)') void add(int x, int y)')
+///   will return 'void add(int _ x, int with y)'
```

Comment:
Why remap the Dart APIs to a hybrid string that will never be used (and has no meaning in either language), only to re-extract the components later, instead of doing the mapping at the point of constructing the Swift code (as is done in the ObjC generator)?

Reply:
I think having this intermediate format will likely make it harder to understand what's happening, and there's more potential for strange side effects (e.g., some new code trying to use the methods, and it not being obvious why they don't match the original source), so I'm concerned about the maintenance impact. I would much rather this transformation happen at the usage point even if it's more code in the short term.

Reply:
Yes, I agree. I'll work on this update.

Reply:
@ailtonvivaz What is the status of this update?

Reply:
Fixed this. Is this approach better?